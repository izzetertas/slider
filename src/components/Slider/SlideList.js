import React from 'react'
import Slide from './Slide'
import './Slider.css'

const SlideList = ({ items }) => {
  if(!items) return null

  return (
    <div className='slider-content'>
    {
      items.map((item, i) => (
        <Slide content={item} key={i}/>
      ))
    }
    </div>
  )
}

export default SlideList
