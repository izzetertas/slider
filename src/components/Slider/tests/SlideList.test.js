import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import SlideList from '../SlideList'


describe('<SlideList/>', () => {
  it('should render correctly with items', () => {
    const wrapper = mount(<SlideList items={['1', '2', '3']} />)
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('div').length).toBe(4)
  })

  it('should render correctly without any items', () => {
    const wrapper = shallow(<SlideList />)
    expect(wrapper.find('div').length).toBe(0)
  })

})