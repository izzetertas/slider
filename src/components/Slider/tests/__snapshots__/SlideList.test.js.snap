// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideList/> should render correctly with items 1`] = `
<SlideList
  items={
    Array [
      "1",
      "2",
      "3",
    ]
  }
>
  <div
    className="slider-content"
  >
    <Slide
      content="1"
      key="0"
    >
      <div
        className="slider-item"
      >
        1
      </div>
    </Slide>
    <Slide
      content="2"
      key="1"
    >
      <div
        className="slider-item"
      >
        2
      </div>
    </Slide>
    <Slide
      content="3"
      key="2"
    >
      <div
        className="slider-item"
      >
        3
      </div>
    </Slide>
  </div>
</SlideList>
`;
